edb/schema/pointers.py

```diff
@@ -452,7 +499,11 @@ def allow_ref_propagation(
         constext: sd.CommandContext,
         refdict: so.RefDict,
     ) -> bool:
-        return not self.get_source(schema).is_view(schema)
+        from edb.schema import objtypes as s_objtypes
+
+        object_type = self.get_source(schema)
+        assert isinstance(object_type, s_objtypes.ObjectType)
```

Comment:
`s_types.Type` would be enough, you don't need to go as narrow as `ObjectType` here